src/OpenLoco/src/GameCommands/RaiseLowerLand.cpp

```diff
@@ -0,0 +1,485 @@
+            0x2C, 0x00, 0x00, 0x00, 0x2C, 0x04, 0x04, 0x07, 0x2C, 0x08, 0x08, 0x0B, 0x2C, 0x0C, 0x0C, 0x0C,
+        },
+        {
+            0x29, 0x29, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x29, 0x29, 0x08, 0x09, 0x08, 0x09, 0x08, 0x09,
+            0x29, 0x29, 0x00, 0x01, 0x00, 0x01, 0x00, 0x07, 0x29, 0x29, 0x08, 0x09, 0x08, 0x09, 0x0E, 0x09,
+        },
+    };
+    // clang-format on
+
+    // 0x00468651
+    // TODO: move to TileManager?
+    static uint32_t adjustSurfaceHeight(World::Pos2 pos, SmallZ targetBaseZ, uint8_t slopeFlags, std::set<World::Pos3, LessThanPos3>& removedBuildings, uint8_t flags)
+    {
+        if (!validCoords(pos))
+        {
+            setErrorText(StringIds::off_edge_of_map);
+            return FAILURE;
+        }
+
+        if (targetBaseZ < 4)
+        {
+            setErrorText(StringIds::error_too_low);
+            return FAILURE;
+        }
+        if (targetBaseZ > 160
+            || (targetBaseZ == 160 && (slopeFlags & 0x1F) == 0)
+            || (targetBaseZ == 156 && (slopeFlags & 0x10) == 0))
+        {
+            setErrorText(StringIds::error_too_high);
+            return FAILURE;
+        }
+
+        currency32_t totalCost = 0;
+
+        if (flags & Flags::apply)
+        {
+            TileManager::removeSurfaceIndustry(pos);
+            TileManager::setTerrainStyleAsCleared(pos);
+
+            auto clearHeight = TileManager::getHeight(pos).landHeight;
+            TileManager::removeAllWallsOnTile(toTileSpace(pos), clearHeight);
+        }
+
+        // Compute cost of landscape operation
+        auto tile = TileManager::get(pos);
+        auto* surface = tile.surface();
+        auto landObj = ObjectManager::get<LandObject>(surface->terrain());
+        totalCost = totalCost + Economy::getInflationAdjustedCost(landObj->costFactor, landObj->costIndex, 10);
+
+        // If surface is in use as a water(route), and has water, ensure we don't remove it
+        if (surface->hasType6Flag() && surface->water() > 0)
+        {
+            auto water = surface->water() * 2 - kMicroToSmallZStep;
+            auto testHeight = targetBaseZ;
+            if (slopeFlags & 0x1F)
+            {
+                testHeight += kSmallZStep;
+                if (slopeFlags & SurfaceSlope::doubleHeight)
+                {
+                    testHeight += kSmallZStep;
+                }
+            }
+
+            if (water < testHeight)
+            {
+                setErrorText(StringIds::water_channel_currently_needed_by_ships);
+                return FAILURE;
+            }
+        }
+
+        // Bind our local vars to the tile clear function
+        auto clearFunc = [pos, &removedBuildings, flags, &totalCost](TileElement& el) {
+            return TileClearance::tileClearFunction(el, pos, removedBuildings, flags, totalCost);
+        };
+
+        QuarterTile qt(0xF, 0);
+        auto tileHeight = TileManager::getHeight(pos);
```

Comment:
getHeight gets the height at pos we need to get the max height for the tile which means:
           ``` auto testHeight = targetBaseZ;
            if (slopeFlags & 0x1F)
            {
                testHeight += kSmallZStep;
                if (slopeFlags & SurfaceSlope::doubleHeight)
                {
                    testHeight += kSmallZStep;
                }
            }```

Reply:
I'm not finding this back in the original function. Is that correct?

Reply:
This is 0x00468757 which i'm saying should be a mini function as its used in  lots of code.